libs/libskynet/src/encrypt.ts

```diff
@@ -0,0 +1,53 @@
+import { encodeU64 } from "./encoding.js";
+import { SHA512_HASH_SIZE, sha512 } from "./sha512.js";
+
+// otpEncrypt takes a key and some data and encrypts the data with the key. The
+// encryption happens by generating a sequence of bytes using sha512 hashes and
+// then xor'ing those bytes with the data. This gives otpEncrypt similar
+// security properties to a one-time-pad - which means that the same key cannot
+// be used twice!
+//
+// It also means that there is no authentication on the data, and that an
+// attacker could flip bits undetected if an authentication layer is not added
+// on top.
+//
+// Data is encrypted in-place. The optional value 'skip' allows the caller to
+// specify a number of bytes to skip initially.
+//
+// NOTE: otpEncrypt can be useful over other encryption methods because it does
+// not introduce any new dependencies. For the Skynet Kernel bootloader, the
+// only cryptography present is ed25519 signatures (which includes sha512 as a
+// dependency). This is a tiny piece of code that can provide encryption
+// support without needing to add a full encryption library as a dependency.
+//
+// WARNING: otpEncrypt is not a "safe" function. It's a useful cryptographic
+// primitive, but there are easy ways to misuse it and create insecure
+// encryption schemes. Please avoid using this function unless you have a
+// strong understanding of encryption techniques and typical encryption
+// attacks.
+function otpEncrypt(key: Uint8Array, data: Uint8Array, skip = 0): Uint8Array {
+  // Build an array to hold the preimage for each step of encryption. We are
+  // just going to be altering the final 8 bytes as we encrypt the file.
+  const preimageHolder = new Uint8Array(key.length + 8);
+  preimageHolder.set(key, 0);
+
+  // Iterate over the data and encrypt each section.
+  for (let i = skip; i < data.length; i += SHA512_HASH_SIZE) {
+    // Set the nonce for this shard and then create the pad data.
+    const [iBytes] = encodeU64(BigInt(i));
+    preimageHolder.set(iBytes, key.length);
+    const keyData = sha512(preimageHolder);
```

Comment:
error is ignored on `encodeU64` because the only possible error is that `data` is larger than 2^64 bytes, which we can be confident won't trigger.

Reply:
I don't like how this process is based on dev understanding the function and assessing whether it's ok to ignore error handling or not.

Reply:
we only do it here because it's a low level function, and that saves error checking on anyone that calls it. In general I don't like it either and it's discouraged, this is one exception because the complexity to justify it isn't much, and the savings are pretty good.

If javascript had a stronger typing system we could make this error at the compiler level if the invariant ever broke.